src/main/java/it/gov/pagopa/admissibility/service/onboarding/AuthoritiesDataRetrieverServiceImpl.java

```diff
@@ -29,101 +36,132 @@ public class AuthoritiesDataRetrieverServiceImpl implements AuthoritiesDataRetri
+                                .build());
+                    }
+                });
+    }
+
+    private Mono<OnboardingDTO> retrieveIsee(OnboardingDTO onboardingRequest, InitiativeConfig initiativeConfig) {
+        return this.retrieveUserIsee(onboardingRequest.getUserId())
+                .switchIfEmpty(mockIsee(onboardingRequest))
+                .doOnNext(m -> setIseeIfCorrespondingType(onboardingRequest, initiativeConfig, m))
+                .map(m -> {
+                    CriteriaCodeConfig criteriaCodeConfig = criteriaCodeService.getCriteriaCodeConfig(OnboardingConstants.CRITERIA_CODE_ISEE);
+
+                    if (onboardingRequest.getIsee() == null) {
+                        throw new OnboardingException(
+                                List.of(new OnboardingRejectionReason(
+                                        OnboardingRejectionReason.OnboardingRejectionReasonType.ISEE_TYPE_KO,
+                                        OnboardingConstants.REJECTION_REASON_ISEE_TYPE_KO,
+                                        criteriaCodeConfig.getAuthority(),
+                                        criteriaCodeConfig.getAuthorityLabel(),
+                                        "ISEE non disponibile"
+                                )),
+                                "User having id %s has not compatible type for initiative %s".formatted(onboardingRequest.getUserId(), initiativeConfig.getInitiativeId()));
+                    }
+
+                    return onboardingRequest;
+                });
+    }
 
-        if (onboardingRequest.getResidence() == null && is2retrieve(initiativeConfig, OnboardingConstants.CRITERIA_CODE_RESIDENCE)) {
-            onboardingRequest.setResidence(
-                    userIdBasedIntegerGenerator(onboardingRequest).nextInt(0, 2) == 0
-                            ? Residence.builder()
-                            .city("Milano")
-                            .cityCouncil("Milano")
-                            .province("Milano")
-                            .region("Lombardia")
-                            .postalCode("20124")
-                            .nation("Italia")
-                            .build()
-                            : Residence.builder()
-                            .city("Roma")
-                            .cityCouncil("Roma")
-                            .province("Roma")
-                            .region("Lazio")
-                            .postalCode("00187")
-                            .nation("Italia")
-                            .build()
-
-            );
+    private Mono<Map<String, BigDecimal>> mockIsee(OnboardingDTO onboardingRequest) {
+        Map<String, BigDecimal> iseeMockMap = new HashMap<>();
+        List<IseeTypologyEnum> iseeList = Arrays.asList(IseeTypologyEnum.values());
+
+        int randomTypology = new Random(onboardingRequest.getUserId().hashCode()).nextInt(1, iseeList.size());
```

Comment:
You are excluding first element
```suggestion
        int randomTypology = new Random(onboardingRequest.getUserId().hashCode()).nextInt(0, iseeList.size());
```